@here/harp-mapview/lib/geometry/TileGeometryLoader.ts

```diff
@@ -246,16 +256,33 @@ export class SimpleTileGeometryLoader implements TileGeometryLoader {
         enabledKinds: GeometryKindSet | undefined,
         disabledKinds: GeometryKindSet | undefined
     ) {
+        if (this.isFinished) {
+            return;
+        }
         // If the tile is not ready for display, or if it has become invisible while being loaded,
         // for example by moving the camera, the tile is not finished and its geometry is not
         // created. This is an optimization for fast camera movements and zooms.
         const tile = this.tile;
         const decodedTile = this.m_decodedTile;
         this.m_decodedTile = undefined;
+        // TODO: Entire block could be moved to update method
         if (decodedTile === undefined || tile.disposed || !tile.isVisible) {
+            // TODO: Should we clearTimer here?
+            if (this.m_timeout !== undefined) {
+                clearTimeout(this.m_timeout);
+            }
+            // TODO: Should we dispose tile here?
+            //if (!tile.isVisible) {
```

Comment:
This is the block that could be moved to update() function in order to cancel loading (timeout) unnecessary tiles immediately, otherwise once loading started it will happen when timeout is processed.